src/on-chain-value.ts

```diff
@@ -27,48 +26,73 @@ export const chainValueDataPoint: APIGatewayProxyHandler = async (event): Promis
     };
   }
 
-  const provider = config.providers[chainId];
-  if (!provider) {
+  const db = await initDb();
+  if(db === undefined) {
     return {
       statusCode: 500,
       headers: config.headers,
-      body: makeError("We don't have a provider for that chainId"),
+      body: makeError("An error has occurred while trying to initialize the database"),
     };
   }
 
-  const dapiServerAddress = contracts.DapiServer[chainId];
-  if (!dapiServerAddress) {
+  var currentDataFeedId = dataFeedId;
+  if(dapiName && !dataFeedId) {
+    const hashedDapiNameId = ethers.utils.formatBytes32String(dapiName);
+    
+    const operation = async () => db.query(`
+      SELECT event_data->'parsedLog'->'args'->> 1 as "dataFeedId" 
+      FROM dapi_events 
+      WHERE 
+      event_name = 'SetDapiName' AND 
+      event_data->'parsedLog'->'args'->> 0 = $1 
+      ORDER BY block DESC
+      LIMIT 1;
+    `, [hashedDapiNameId]);
+
+    const [err, firstResult] = await go(operation, { timeoutMs: 5_000, retries: 2 });
+    if (err || firstResult === undefined) {
+      return {
+        statusCode: 500,
+        headers: config.headers,
+        body: makeError("An error has occurred while querying the dataFeedId"),
+      };
+    }
+    currentDataFeedId = firstResult.rows[0].dataFeedId;
+  }
+
+  const operation = async () => db.query(`
+      SELECT event_data -> 'parsedLog' -> 'args' ->> 1 as "data"
+      FROM dapi_events 
+      WHERE 
+      chain = $1 AND
+      event_data->'parsedLog'->'args'->>0 = $2
+      ORDER by block DESC 
+      LIMIT 1;
+    `, [chainId, currentDataFeedId]
+  );
+
+  const [err, queryResult] = await go(operation, { timeoutMs: 5_000, retries: 2 });
+  if (err || queryResult === undefined) {
```

Comment:
I suggest that you print the error to the console so we can debug later, eg. `console.error(...`